web-console/src/lib/components/streaming/management/PipelineTable.tsx

```diff
@@ -574,23 +574,27 @@ export default function PipelineTable() {
 }
 
 const usePipelineStatus = (params: { row: Pipeline }) => {
-  const pipeline = params.row.descriptor
+  const { data: pipelines } = useQuery({
+    ...PipelineManagerQuery.pipelines()
+  })
+  const pipelineData = pipelines?.find(p => p.descriptor.pipeline_id === params.row.descriptor.pipeline_id)
+  const pipeline = pipelineData?.descriptor
+  invariant(pipeline, 'Pipeline should be found')
```

Comment:
this is mostly a side note something to fix for the future: I think invariants are great, but in practice they still trigger sometimes, so it would be great to have a better failure mechanism for them at least in production builds e.g., I think in dev having an assert() is fine, but in production ideally it would display an error box instead of asserts which results in just a white screen which says  'things have crashed' and isn't very useful?

Reply:
I like invariants - I use them when I want a check that should NEVER EVER fail (and not take it on faith in the logic). So whenever there is a fail - it's obtrusive and forces me to find and fix the root of the issue. For reasonable runtime errors I do use throws that result in an error box.

Reply:
I like invariants too, I'm not saying we shouldnt get rid of them. But if the behavior is that the whole app is unusable e.g., the menu disappears then ideally this doesn't have to be the case (e.g., if some components menu can continue to function correctly, we should keep them around for the user to click)